BUILD.md

```diff
@@ -0,0 +1,472 @@
+:balloon: Welcome to the Conduit development guide! :wave:
+
+This document will help you build Conduit from source. Depending on the
+components you are interested in building, these instructions assume you have
+the following tools already setup:
+- [Go](https://golang.org)
+- [Rust](https://www.rust-lang.org)
+- [Minikube](https://github.com/kubernetes/minikube)
+- [Docker](https://www.docker.com/)
+- [Yarn](https://yarnpkg.com)
```

Comment:
I think it would be better to reference each of these dependencies in the specific sections, with links/descriptions to our recommended install instructions for each.

Having this at the top makes me think i have to go install all of these things first and foremost, which isn't true.

Reply:
Makes sense. I've re-org'd the doc around Go/Web/Rust/Release, with each section documenting its own dependencies, build, and test commands.